From patchwork Tue Dec  3 16:58:58 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Xu <peterx@redhat.com>
X-Patchwork-Id: 11271713
Return-Path: <SRS0=eZxF=ZZ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 711F5112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:41 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 4EBF620803
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:41 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="Zn08heVy"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727192AbfLCQ7M (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 3 Dec 2019 11:59:12 -0500
Received: from us-smtp-delivery-1.mimecast.com ([205.139.110.120]:49293 "EHLO
        us-smtp-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org
        with ESMTP id S1727154AbfLCQ7K (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 3 Dec 2019 11:59:10 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1575392349;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=kn/llSEPws9Kaihse6HkAcboPO6kvjhOBVqBu50kJWM=;
        b=Zn08heVy4i5cwuaWKecR8YKjhfSjACBEP5Q7tINwU0llVC8rSIY/9fpsPguox7yFDB3lsc
        rNgHpbtgxDq1m5L84HJvilI/vyLi3cKCLmKxYLaZ43LH73JevyiECp9rPPcNwNkMf5iYEb
        d7tzT6H+qAsTrNCRuPIigqdgeBUy5qA=
Received: from mail-qv1-f69.google.com (mail-qv1-f69.google.com
 [209.85.219.69]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-114-T-dleMvAO1eVjNjfE7xNJQ-1; Tue, 03 Dec 2019 11:59:08 -0500
Received: by mail-qv1-f69.google.com with SMTP id l1so2583748qvu.13
        for <kvm@vger.kernel.org>; Tue, 03 Dec 2019 08:59:07 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=qDMFszQpTCz8aKUy2p2BRtrOo8b1SfrY7qPPp6I+jnk=;
        b=g+W/jA/3/Wg4aDK+YFGj0TnyNBssjR3is37PfTMbkoQbnyM1gFXTUYhuq8UyaL+mMI
         SXiVHb/v1PYORtscaxOSD2eCuNyG7fD4mW8C0hZgOsyBuLPOOw84fMCRdauhwW0mnASZ
         3RttCGOehUeVc34b/cWd0N2qQZJcCXBbNkJcMX2sZ/9PsmseFppeoSGQoe6zJH2wRHaE
         R1w1fZXCECRTd6wn7o/sKKcZvQFqN7ROOt3y7HVtL2H/jsyrmWylWRbt7p03ugEeShbT
         iGwH594YBNCEWcFZTyDnNy4MJjNmVr6kGolOKacRi7OxJRYJb9mNfoaFoqc907tv/2r3
         OiVA==
X-Gm-Message-State: APjAAAVNM1VQHttqzRXnOMGkoXACtceXSOa2EXrWVacYYYsa1jMLr6Sd
        baQfPWX+xhrfONo7Q3eHh9o09cOKlkiPGBfy2Jjm5WiAUXSnHJmxt60YrLwI6uY+2OdMWsSq3AZ
        16ZyIcbkyoI/u
X-Received: by 2002:ac8:1017:: with SMTP id z23mr5821776qti.94.1575392347081;
        Tue, 03 Dec 2019 08:59:07 -0800 (PST)
X-Google-Smtp-Source: 
 APXvYqxg/JOE/OtM1jHS/faP8zXtfiNcD6HKzy7U3+/q3M6bfRuQ5NZYhl14b7ItPtyjdu5wi5+3bg==
X-Received: by 2002:ac8:1017:: with SMTP id z23mr5821756qti.94.1575392346887;
        Tue, 03 Dec 2019 08:59:06 -0800 (PST)
Received: from xz-x1.yyz.redhat.com ([104.156.64.74])
        by smtp.gmail.com with ESMTPSA id
 a16sm482585qkn.48.2019.12.03.08.59.05
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Dec 2019 08:59:05 -0800 (PST)
From: Peter Xu <peterx@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Cc: Nitesh Narayan Lal <nitesh@redhat.com>,
        Paolo Bonzini <pbonzini@redhat.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        peterx@redhat.com, Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [PATCH v4 1/6] KVM: X86: Fix kvm_bitmap_or_dest_vcpus() to use irq
 shorthand
Date: Tue,  3 Dec 2019 11:58:58 -0500
Message-Id: <20191203165903.22917-2-peterx@redhat.com>
X-Mailer: git-send-email 2.21.0
In-Reply-To: <20191203165903.22917-1-peterx@redhat.com>
References: <20191203165903.22917-1-peterx@redhat.com>
MIME-Version: 1.0
X-MC-Unique: T-dleMvAO1eVjNjfE7xNJQ-1
X-Mimecast-Spam-Score: 0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The 3rd parameter of kvm_apic_match_dest() is the irq shorthand,
rather than the irq delivery mode.

Fixes: 7ee30bc132c6 ("KVM: x86: deliver KVM IOAPIC scan request to target vCPUs")
Reviewed-by: Vitaly Kuznetsov <vkuznets@redhat.com>
Signed-off-by: Peter Xu <peterx@redhat.com>
---
 arch/x86/kvm/lapic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kvm/lapic.c b/arch/x86/kvm/lapic.c
index cf9177b4a07f..1eabe58bb6d5 100644
--- a/arch/x86/kvm/lapic.c
+++ b/arch/x86/kvm/lapic.c
@@ -1151,7 +1151,7 @@ void kvm_bitmap_or_dest_vcpus(struct kvm *kvm, struct kvm_lapic_irq *irq,
 			if (!kvm_apic_present(vcpu))
 				continue;
 			if (!kvm_apic_match_dest(vcpu, NULL,
-						 irq->delivery_mode,
+						 irq->shorthand,
 						 irq->dest_id,
 						 irq->dest_mode))
 				continue;

From patchwork Tue Dec  3 16:58:59 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Xu <peterx@redhat.com>
X-Patchwork-Id: 11271711
Return-Path: <SRS0=eZxF=ZZ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id C9606112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:39 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 9E82D2084B
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:39 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="Ksxictvd"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727207AbfLCQ7M (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 3 Dec 2019 11:59:12 -0500
Received: from us-smtp-1.mimecast.com ([205.139.110.61]:40136 "EHLO
        us-smtp-delivery-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL)
        by vger.kernel.org with ESMTP id S1727159AbfLCQ7M (ORCPT
        <rfc822;kvm@vger.kernel.org>); Tue, 3 Dec 2019 11:59:12 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1575392350;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=5z9DSSJjWBkwMoPX0AiQMG2lTnPniK2p7yuPt+JjwOw=;
        b=KsxictvdNx9MTScpIwVs/ACT/BgSQ7btAVjU/wak6ALfrMo28pHEsqc1uWdXfDEVQqc9LB
        pD8EnYGlFVv3UrUmZ7TwhX1Stw9XdJAKPUIg6ZNuReuYQyYqt/ZL7b73JTllHLNkvkEVzX
        4IoM3ttGgRr2HQrN37UCTwb3I6yg2IY=
Received: from mail-qv1-f70.google.com (mail-qv1-f70.google.com
 [209.85.219.70]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-162-7mEpew6eMCq7HFS74tO4LQ-1; Tue, 03 Dec 2019 11:59:09 -0500
Received: by mail-qv1-f70.google.com with SMTP id q17so721540qvo.23
        for <kvm@vger.kernel.org>; Tue, 03 Dec 2019 08:59:09 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=v2wRBhbJMNEdghlygF10/9hA60Iod/Z62OhKSAkteiI=;
        b=kDlU9t/GRiaN8Rq30kg7YpVifXNBkZmvCQCoi7yIU5hj/jieWLdDlm3JSnKxHftm88
         6424yTVo/puv9rEH//0t+3Wjlv6hPjSff7YyH009JDhfOYjGMc+lyemS8BLDiUFhyXOe
         eqJ1KtLT5IbvFkaHQHEZf/usen+QG7qQc+o9QBfCR42ORTvv3jlpODHFLF5+bexHvUZx
         aZCLNcv1ymwOF+RrO8Y+ihJrUy2b0lke+3Q/D0WjAti7wr3SolgCPNW8w2lmK9OeYdHH
         WjMGP7PIb4oeNhq1qJC5m0srKstKtG0e2rOwp7xp9rQIimvPk5lPKERv2bRzjg7tMjJI
         T16Q==
X-Gm-Message-State: APjAAAXksGGXG/vyPYnzqbwr3Ij5GqYBPH7wFd/6M6iG9zDEeNSsBHks
        1nj8F/7TbQ9CaBVrGhap3Q11lWoddqkRgEMyhqNngeUWh3dDBs9+rrJgbnlVwbrEVY75QRhUDP9
        pl1bXwY7k+5gT
X-Received: by 2002:a37:a3c1:: with SMTP id m184mr6052522qke.49.1575392348558;
        Tue, 03 Dec 2019 08:59:08 -0800 (PST)
X-Google-Smtp-Source: 
 APXvYqyG8wKpsZ7GEJ5gyhStD0FO722TIQr2qgwbQ3M/iVrKE5wtsIGvlcXpXI0BPSfAtXE+sbQMSg==
X-Received: by 2002:a37:a3c1:: with SMTP id m184mr6052500qke.49.1575392348302;
        Tue, 03 Dec 2019 08:59:08 -0800 (PST)
Received: from xz-x1.yyz.redhat.com ([104.156.64.74])
        by smtp.gmail.com with ESMTPSA id
 a16sm482585qkn.48.2019.12.03.08.59.06
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Dec 2019 08:59:07 -0800 (PST)
From: Peter Xu <peterx@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Cc: Nitesh Narayan Lal <nitesh@redhat.com>,
        Paolo Bonzini <pbonzini@redhat.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        peterx@redhat.com, Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [PATCH v4 2/6] KVM: X86: Move irrelevant declarations out of ioapic.h
Date: Tue,  3 Dec 2019 11:58:59 -0500
Message-Id: <20191203165903.22917-3-peterx@redhat.com>
X-Mailer: git-send-email 2.21.0
In-Reply-To: <20191203165903.22917-1-peterx@redhat.com>
References: <20191203165903.22917-1-peterx@redhat.com>
MIME-Version: 1.0
X-MC-Unique: 7mEpew6eMCq7HFS74tO4LQ-1
X-Mimecast-Spam-Score: 0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

kvm_apic_match_dest() is declared in both ioapic.h and lapic.h.
Remove the declaration in ioapic.h.

kvm_apic_compare_prio() is declared in ioapic.h but defined in
lapic.c.  Move the declaration to lapic.h.

kvm_irq_delivery_to_apic() is declared in ioapic.h but defined in
irq_comm.c.  Move the declaration to irq.h.

hyperv.c needs to use kvm_irq_delivery_to_apic(). Include irq.h in
hyperv.c.

Reviewed-by: Vitaly Kuznetsov <vkuznets@redhat.com>
Signed-off-by: Peter Xu <peterx@redhat.com>
---
 arch/x86/kvm/hyperv.c | 1 +
 arch/x86/kvm/ioapic.h | 6 ------
 arch/x86/kvm/irq.h    | 3 +++
 arch/x86/kvm/lapic.h  | 2 +-
 4 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/arch/x86/kvm/hyperv.c b/arch/x86/kvm/hyperv.c
index 23ff65504d7e..c7d4640b7b1c 100644
--- a/arch/x86/kvm/hyperv.c
+++ b/arch/x86/kvm/hyperv.c
@@ -33,6 +33,7 @@
 #include <trace/events/kvm.h>
 
 #include "trace.h"
+#include "irq.h"
 
 #define KVM_HV_MAX_SPARSE_VCPU_SET_BITS DIV_ROUND_UP(KVM_MAX_VCPUS, 64)
 
diff --git a/arch/x86/kvm/ioapic.h b/arch/x86/kvm/ioapic.h
index ea1a4e0297da..2fb2e3c80724 100644
--- a/arch/x86/kvm/ioapic.h
+++ b/arch/x86/kvm/ioapic.h
@@ -116,9 +116,6 @@ static inline int ioapic_in_kernel(struct kvm *kvm)
 }
 
 void kvm_rtc_eoi_tracking_restore_one(struct kvm_vcpu *vcpu);
-bool kvm_apic_match_dest(struct kvm_vcpu *vcpu, struct kvm_lapic *source,
-		int short_hand, unsigned int dest, int dest_mode);
-int kvm_apic_compare_prio(struct kvm_vcpu *vcpu1, struct kvm_vcpu *vcpu2);
 void kvm_ioapic_update_eoi(struct kvm_vcpu *vcpu, int vector,
 			int trigger_mode);
 int kvm_ioapic_init(struct kvm *kvm);
@@ -126,9 +123,6 @@ void kvm_ioapic_destroy(struct kvm *kvm);
 int kvm_ioapic_set_irq(struct kvm_ioapic *ioapic, int irq, int irq_source_id,
 		       int level, bool line_status);
 void kvm_ioapic_clear_all(struct kvm_ioapic *ioapic, int irq_source_id);
-int kvm_irq_delivery_to_apic(struct kvm *kvm, struct kvm_lapic *src,
-			     struct kvm_lapic_irq *irq,
-			     struct dest_map *dest_map);
 void kvm_get_ioapic(struct kvm *kvm, struct kvm_ioapic_state *state);
 void kvm_set_ioapic(struct kvm *kvm, struct kvm_ioapic_state *state);
 void kvm_ioapic_scan_entry(struct kvm_vcpu *vcpu,
diff --git a/arch/x86/kvm/irq.h b/arch/x86/kvm/irq.h
index 7c6233d37c64..f173ab6b407e 100644
--- a/arch/x86/kvm/irq.h
+++ b/arch/x86/kvm/irq.h
@@ -113,5 +113,8 @@ int apic_has_pending_timer(struct kvm_vcpu *vcpu);
 
 int kvm_setup_default_irq_routing(struct kvm *kvm);
 int kvm_setup_empty_irq_routing(struct kvm *kvm);
+int kvm_irq_delivery_to_apic(struct kvm *kvm, struct kvm_lapic *src,
+			     struct kvm_lapic_irq *irq,
+			     struct dest_map *dest_map);
 
 #endif
diff --git a/arch/x86/kvm/lapic.h b/arch/x86/kvm/lapic.h
index 39925afdfcdc..0b9bbadd1f3c 100644
--- a/arch/x86/kvm/lapic.h
+++ b/arch/x86/kvm/lapic.h
@@ -83,7 +83,7 @@ int kvm_lapic_reg_read(struct kvm_lapic *apic, u32 offset, int len,
 		       void *data);
 bool kvm_apic_match_dest(struct kvm_vcpu *vcpu, struct kvm_lapic *source,
 			   int short_hand, unsigned int dest, int dest_mode);
-
+int kvm_apic_compare_prio(struct kvm_vcpu *vcpu1, struct kvm_vcpu *vcpu2);
 bool __kvm_apic_update_irr(u32 *pir, void *regs, int *max_irr);
 bool kvm_apic_update_irr(struct kvm_vcpu *vcpu, u32 *pir, int *max_irr);
 void kvm_apic_update_ppr(struct kvm_vcpu *vcpu);

From patchwork Tue Dec  3 16:59:00 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Xu <peterx@redhat.com>
X-Patchwork-Id: 11271707
Return-Path: <SRS0=eZxF=ZZ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id AACB16C1
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:32 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 7EEF420803
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:32 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="BVghTG4T"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726928AbfLCQ7b (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 3 Dec 2019 11:59:31 -0500
Received: from us-smtp-2.mimecast.com ([205.139.110.61]:37364 "EHLO
        us-smtp-delivery-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL)
        by vger.kernel.org with ESMTP id S1727166AbfLCQ7P (ORCPT
        <rfc822;kvm@vger.kernel.org>); Tue, 3 Dec 2019 11:59:15 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1575392354;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=9PzwZcxUy7RnohkGRP3nmuFGuUFWbAoxrN1OoQL2mAM=;
        b=BVghTG4TqVBI8NkVahcA5mB1gKQW2QJw3cKro7woXtrtdz7BsaXngGMo883t2hXRdTpxfR
        vmtEANm9Whzu+ys2A80j9a3Kw0SHGsvvC5hlGeP34XDROZ+UUjeVP0AHORE1BGYzOkE5Sp
        bzj8xbeNJM1aIAilqUzctcvI59W4zxo=
Received: from mail-qk1-f198.google.com (mail-qk1-f198.google.com
 [209.85.222.198]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-102-IHpHvZPVNs6FTrrBruaMag-1; Tue, 03 Dec 2019 11:59:10 -0500
Received: by mail-qk1-f198.google.com with SMTP id 16so2622153qkm.5
        for <kvm@vger.kernel.org>; Tue, 03 Dec 2019 08:59:10 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=9or0Z3IjCalxRQDLajBB/eFOZ4QHEXiRdq3XJW4DEO4=;
        b=BE4C5T6KNjnwZAA395DUCQaytdSTgnx/HKdbFN4piBqvoB9EMcTwssHm750o3P3zfn
         tLRSfqm9gpqwnJQIqYlNcUR7MPiQs6CSkENphjE0WBr8+jjh7r+1dE6r76Juh4Zd812Y
         fSuBTvqY92sg7YOdZ+nkgAVi4V2XAo94y7cF86FzKuuLRT3Cs+2bgfzvTgdGfkq8gMru
         9bMw2kTPAQ4ezKHLcJGDYwB4fgiUcjvO/+1PjVOEzHmk1R7y1xr3tUNxSbTMBHAgJmUj
         YHPwBPHMmTRqBxhBFVq2IWFhJtQcqMK2OnHaxohJnIscG07CQglkYLSrWpph4SadLTnh
         26UA==
X-Gm-Message-State: APjAAAWn6NvaO5sNZa4gzmH2NBiPLQu5pdYSr2x91DLz4+tP9ER3EiBX
        mX436762mb4D+AkJqPiDv+CenlYoTaMvk5CY6+wdcwHj4HhY816alr7aOeOwJvf/XFCLRCz+lOi
        AW8rNFVbvD9aD
X-Received: by 2002:a37:7bc7:: with SMTP id
 w190mr6056702qkc.132.1575392350173;
        Tue, 03 Dec 2019 08:59:10 -0800 (PST)
X-Google-Smtp-Source: 
 APXvYqxnqxZ1WErwO4sJh61fQ7tyCWYjuH45lvTfKDEeEAS1rAr7HD07UWPgL0K0OGrnA8lJrKUi/w==
X-Received: by 2002:a37:7bc7:: with SMTP id
 w190mr6056671qkc.132.1575392349797;
        Tue, 03 Dec 2019 08:59:09 -0800 (PST)
Received: from xz-x1.yyz.redhat.com ([104.156.64.74])
        by smtp.gmail.com with ESMTPSA id
 a16sm482585qkn.48.2019.12.03.08.59.08
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Dec 2019 08:59:08 -0800 (PST)
From: Peter Xu <peterx@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Cc: Nitesh Narayan Lal <nitesh@redhat.com>,
        Paolo Bonzini <pbonzini@redhat.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        peterx@redhat.com, Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [PATCH v4 3/6] KVM: X86: Use APIC_DEST_* macros properly in
 kvm_lapic_irq.dest_mode
Date: Tue,  3 Dec 2019 11:59:00 -0500
Message-Id: <20191203165903.22917-4-peterx@redhat.com>
X-Mailer: git-send-email 2.21.0
In-Reply-To: <20191203165903.22917-1-peterx@redhat.com>
References: <20191203165903.22917-1-peterx@redhat.com>
MIME-Version: 1.0
X-MC-Unique: IHpHvZPVNs6FTrrBruaMag-1
X-Mimecast-Spam-Score: 0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

We were using either APIC_DEST_PHYSICAL|APIC_DEST_LOGICAL or 0|1 to
fill in kvm_lapic_irq.dest_mode.  It's fine only because in most cases
when we check against dest_mode it's against APIC_DEST_PHYSICAL (which
equals to 0).  However, that's not consistent.  We'll have problem
when we want to start checking against APIC_DEST_LOGICAL, which does
not equals to 1.

This patch firstly introduces kvm_lapic_irq_dest_mode() helper to take
any boolean of destination mode and return the APIC_DEST_* macro.
Then, it replaces the 0|1 settings of irq.dest_mode with the helper.

Signed-off-by: Peter Xu <peterx@redhat.com>
---
 arch/x86/include/asm/kvm_host.h | 5 +++++
 arch/x86/kvm/ioapic.c           | 9 ++++++---
 arch/x86/kvm/irq_comm.c         | 7 ++++---
 arch/x86/kvm/x86.c              | 2 +-
 4 files changed, 16 insertions(+), 7 deletions(-)

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index b79cd6aa4075..f815c97b1b57 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1022,6 +1022,11 @@ struct kvm_lapic_irq {
 	bool msi_redir_hint;
 };
 
+static inline u16 kvm_lapic_irq_dest_mode(bool dest_mode)
+{
+	return dest_mode ? APIC_DEST_LOGICAL : APIC_DEST_PHYSICAL;
+}
+
 struct kvm_x86_ops {
 	int (*cpu_has_kvm_support)(void);          /* __init */
 	int (*disabled_by_bios)(void);             /* __init */
diff --git a/arch/x86/kvm/ioapic.c b/arch/x86/kvm/ioapic.c
index 9fd2dd89a1c5..e623a4f8d27e 100644
--- a/arch/x86/kvm/ioapic.c
+++ b/arch/x86/kvm/ioapic.c
@@ -331,7 +331,8 @@ static void ioapic_write_indirect(struct kvm_ioapic *ioapic, u32 val)
 			irq.vector = e->fields.vector;
 			irq.delivery_mode = e->fields.delivery_mode << 8;
 			irq.dest_id = e->fields.dest_id;
-			irq.dest_mode = e->fields.dest_mode;
+			irq.dest_mode =
+			    kvm_lapic_irq_dest_mode(!!e->fields.dest_mode);
 			bitmap_zero(&vcpu_bitmap, 16);
 			kvm_bitmap_or_dest_vcpus(ioapic->kvm, &irq,
 						 &vcpu_bitmap);
@@ -343,7 +344,9 @@ static void ioapic_write_indirect(struct kvm_ioapic *ioapic, u32 val)
 				 * keep ioapic_handled_vectors synchronized.
 				 */
 				irq.dest_id = old_dest_id;
-				irq.dest_mode = old_dest_mode;
+				irq.dest_mode =
+				    kvm_lapic_irq_dest_mode(
+					!!e->fields.dest_mode);
 				kvm_bitmap_or_dest_vcpus(ioapic->kvm, &irq,
 							 &vcpu_bitmap);
 			}
@@ -369,7 +372,7 @@ static int ioapic_service(struct kvm_ioapic *ioapic, int irq, bool line_status)
 
 	irqe.dest_id = entry->fields.dest_id;
 	irqe.vector = entry->fields.vector;
-	irqe.dest_mode = entry->fields.dest_mode;
+	irqe.dest_mode = kvm_lapic_irq_dest_mode(!!entry->fields.dest_mode);
 	irqe.trig_mode = entry->fields.trig_mode;
 	irqe.delivery_mode = entry->fields.delivery_mode << 8;
 	irqe.level = 1;
diff --git a/arch/x86/kvm/irq_comm.c b/arch/x86/kvm/irq_comm.c
index 8ecd48d31800..22108ed66a76 100644
--- a/arch/x86/kvm/irq_comm.c
+++ b/arch/x86/kvm/irq_comm.c
@@ -52,8 +52,8 @@ int kvm_irq_delivery_to_apic(struct kvm *kvm, struct kvm_lapic *src,
 	unsigned long dest_vcpu_bitmap[BITS_TO_LONGS(KVM_MAX_VCPUS)];
 	unsigned int dest_vcpus = 0;
 
-	if (irq->dest_mode == 0 && irq->dest_id == 0xff &&
-			kvm_lowest_prio_delivery(irq)) {
+	if (irq->dest_mode == APIC_DEST_PHYSICAL &&
+	    irq->dest_id == 0xff && kvm_lowest_prio_delivery(irq)) {
 		printk(KERN_INFO "kvm: apic: phys broadcast and lowest prio\n");
 		irq->delivery_mode = APIC_DM_FIXED;
 	}
@@ -114,7 +114,8 @@ void kvm_set_msi_irq(struct kvm *kvm, struct kvm_kernel_irq_routing_entry *e,
 		irq->dest_id |= MSI_ADDR_EXT_DEST_ID(e->msi.address_hi);
 	irq->vector = (e->msi.data &
 			MSI_DATA_VECTOR_MASK) >> MSI_DATA_VECTOR_SHIFT;
-	irq->dest_mode = (1 << MSI_ADDR_DEST_MODE_SHIFT) & e->msi.address_lo;
+	irq->dest_mode = kvm_lapic_irq_dest_mode(
+	    !!((1 << MSI_ADDR_DEST_MODE_SHIFT) & e->msi.address_lo));
 	irq->trig_mode = (1 << MSI_DATA_TRIGGER_SHIFT) & e->msi.data;
 	irq->delivery_mode = e->msi.data & 0x700;
 	irq->msi_redir_hint = ((e->msi.address_lo
diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 3ed167e039e5..3b00d662dc14 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -7356,7 +7356,7 @@ static void kvm_pv_kick_cpu_op(struct kvm *kvm, unsigned long flags, int apicid)
 	struct kvm_lapic_irq lapic_irq;
 
 	lapic_irq.shorthand = 0;
-	lapic_irq.dest_mode = 0;
+	lapic_irq.dest_mode = APIC_DEST_PHYSICAL;
 	lapic_irq.level = 0;
 	lapic_irq.dest_id = apicid;
 	lapic_irq.msi_redir_hint = false;

From patchwork Tue Dec  3 16:59:01 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Xu <peterx@redhat.com>
X-Patchwork-Id: 11271709
Return-Path: <SRS0=eZxF=ZZ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id D79A5138D
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:32 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id B5BE420803
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:32 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="aOsifXDB"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727297AbfLCQ7c (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 3 Dec 2019 11:59:32 -0500
Received: from us-smtp-delivery-1.mimecast.com ([205.139.110.120]:32177 "EHLO
        us-smtp-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org
        with ESMTP id S1727256AbfLCQ7P (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 3 Dec 2019 11:59:15 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1575392355;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=14g8RaA5piWvH+2ustd/wNI5q+TQc8KxgzZhf+/2kiM=;
        b=aOsifXDBmooHhNfMJWj98QlYRrBcyXzpDfrXEXNMTSMyVK3JwBO+3wHKGlofXbZkrsYL10
        Z8EBhHWQ5DJZDmf5toCOvlaOIQfRdZ3Ip6F5WrghSCGY/ubgoo2JTj90K7Mgw5z2te30yg
        8FYxj2dQa9MotxzvJGlGbMSpsShqjpw=
Received: from mail-qt1-f198.google.com (mail-qt1-f198.google.com
 [209.85.160.198]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-138-FnzV2tSKOb6wZKLshthW6Q-1; Tue, 03 Dec 2019 11:59:12 -0500
Received: by mail-qt1-f198.google.com with SMTP id q17so2870052qtc.8
        for <kvm@vger.kernel.org>; Tue, 03 Dec 2019 08:59:11 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=dMSAhkBc+XDUp7X0JKW/OfiFquvplAZgCqXULfdRF2I=;
        b=XyDSctoB1/cKjo8Nb2eCkHcSaP1jYGiVycA2mcwVWO3sTnh2FrwgFJU9lalbDKs9TH
         slece5wM7lJMcnT1jlASSxDE1xMg/fNWY3iQul32mn/q1X9sMai0r1kKK7Ffbd/jzU4D
         6AKVlBI3UkhRTvctL2O75AhdybUHteMGK1dddIKZStoMk+zMLvKPZPn4Hcms52f4yHei
         ZruV/sWviVZ/y+vj5eB7LXGKBqkC5/qEpxx+RgNIDVSamQH5PYwfOqPE3nhVtR8KhK4e
         B5ER93oJyPiQNGbCMuKAImQjcSoOzCYt3fryjNQQpMiHx+IeeFKokxqnMrgKl60oV2SL
         mHtA==
X-Gm-Message-State: APjAAAUnJ2DmQ4SwWfn/AzFo8YJwXNQxUwXbn5LwNfM6faYiYcKzO5TC
        /eurcDqtOTqMyPaWsN0vx9yQWCerljQ703kWspP1KtaEp0Om1B1FnF49xywd0ggbAtIc3MV77p1
        djhXw+ULOHqde
X-Received: by 2002:ac8:4813:: with SMTP id g19mr6086864qtq.165.1575392351669;
        Tue, 03 Dec 2019 08:59:11 -0800 (PST)
X-Google-Smtp-Source: 
 APXvYqwhOExZF/82Calck5ed4HtF0UB8ns8wW9p0mAS3lY9jroDNugbV1+DN2S5djA6t1ZB7YgfJ5A==
X-Received: by 2002:ac8:4813:: with SMTP id g19mr6086847qtq.165.1575392351452;
        Tue, 03 Dec 2019 08:59:11 -0800 (PST)
Received: from xz-x1.yyz.redhat.com ([104.156.64.74])
        by smtp.gmail.com with ESMTPSA id
 a16sm482585qkn.48.2019.12.03.08.59.09
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Dec 2019 08:59:10 -0800 (PST)
From: Peter Xu <peterx@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Cc: Nitesh Narayan Lal <nitesh@redhat.com>,
        Paolo Bonzini <pbonzini@redhat.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        peterx@redhat.com, Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [PATCH v4 4/6] KVM: X86: Drop KVM_APIC_SHORT_MASK and
 KVM_APIC_DEST_MASK
Date: Tue,  3 Dec 2019 11:59:01 -0500
Message-Id: <20191203165903.22917-5-peterx@redhat.com>
X-Mailer: git-send-email 2.21.0
In-Reply-To: <20191203165903.22917-1-peterx@redhat.com>
References: <20191203165903.22917-1-peterx@redhat.com>
MIME-Version: 1.0
X-MC-Unique: FnzV2tSKOb6wZKLshthW6Q-1
X-Mimecast-Spam-Score: 0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

We have both APIC_SHORT_MASK and KVM_APIC_SHORT_MASK defined for the
shorthand mask.  Similarly, we have both APIC_DEST_MASK and
KVM_APIC_DEST_MASK defined for the destination mode mask.

Drop the KVM_APIC_* macros and replace the only user of them to use
the APIC_DEST_* macros instead.  At the meantime, move APIC_SHORT_MASK
and APIC_DEST_MASK from lapic.c to lapic.h.

Reviewed-by: Vitaly Kuznetsov <vkuznets@redhat.com>
Signed-off-by: Peter Xu <peterx@redhat.com>
---
 arch/x86/kvm/lapic.c | 3 ---
 arch/x86/kvm/lapic.h | 5 +++--
 arch/x86/kvm/svm.c   | 4 ++--
 3 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/arch/x86/kvm/lapic.c b/arch/x86/kvm/lapic.c
index 1eabe58bb6d5..805c18178bbf 100644
--- a/arch/x86/kvm/lapic.c
+++ b/arch/x86/kvm/lapic.c
@@ -56,9 +56,6 @@
 #define APIC_VERSION			(0x14UL | ((KVM_APIC_LVT_NUM - 1) << 16))
 #define LAPIC_MMIO_LENGTH		(1 << 12)
 /* followed define is not in apicdef.h */
-#define APIC_SHORT_MASK			0xc0000
-#define APIC_DEST_NOSHORT		0x0
-#define APIC_DEST_MASK			0x800
 #define MAX_APIC_VECTOR			256
 #define APIC_VECTORS_PER_REG		32
 
diff --git a/arch/x86/kvm/lapic.h b/arch/x86/kvm/lapic.h
index 0b9bbadd1f3c..5a9f29ed9a4b 100644
--- a/arch/x86/kvm/lapic.h
+++ b/arch/x86/kvm/lapic.h
@@ -10,8 +10,9 @@
 #define KVM_APIC_SIPI		1
 #define KVM_APIC_LVT_NUM	6
 
-#define KVM_APIC_SHORT_MASK	0xc0000
-#define KVM_APIC_DEST_MASK	0x800
+#define APIC_SHORT_MASK			0xc0000
+#define APIC_DEST_NOSHORT		0x0
+#define APIC_DEST_MASK			0x800
 
 #define APIC_BUS_CYCLE_NS       1
 #define APIC_BUS_FREQUENCY      (1000000000ULL / APIC_BUS_CYCLE_NS)
diff --git a/arch/x86/kvm/svm.c b/arch/x86/kvm/svm.c
index 362e874297e4..65a27a7e9cb1 100644
--- a/arch/x86/kvm/svm.c
+++ b/arch/x86/kvm/svm.c
@@ -4519,9 +4519,9 @@ static int avic_incomplete_ipi_interception(struct vcpu_svm *svm)
 		 */
 		kvm_for_each_vcpu(i, vcpu, kvm) {
 			bool m = kvm_apic_match_dest(vcpu, apic,
-						     icrl & KVM_APIC_SHORT_MASK,
+						     icrl & APIC_SHORT_MASK,
 						     GET_APIC_DEST_FIELD(icrh),
-						     icrl & KVM_APIC_DEST_MASK);
+						     icrl & APIC_DEST_MASK);
 
 			if (m && !avic_vcpu_is_running(vcpu))
 				kvm_vcpu_wake_up(vcpu);

From patchwork Tue Dec  3 16:59:02 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Xu <peterx@redhat.com>
X-Patchwork-Id: 11271705
Return-Path: <SRS0=eZxF=ZZ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 5DA7F138D
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:25 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 3D04D2080A
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:25 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="WWUihS3+"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727283AbfLCQ7R (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 3 Dec 2019 11:59:17 -0500
Received: from us-smtp-delivery-1.mimecast.com ([205.139.110.120]:23956 "EHLO
        us-smtp-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org
        with ESMTP id S1727247AbfLCQ7Q (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 3 Dec 2019 11:59:16 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1575392354;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=EbbDA8eBzmrrmvpNYXesODqmal32f0agXVmci0jBwiY=;
        b=WWUihS3+FbM84emteup784ANuWqQR/6Dn/+94P7F9FH9t3qZ3CgCHRaDdH+O6qIHmBMu7n
        Znihz96exwCo8qdzHneZG+2JNaVysIM2Ceu7ew7RWuuu8JH78C8+3O0fSNA0D1NkjdYMUa
        YyjheS+ooBG33dVhOgTA2d/+wvP+rFs=
Received: from mail-qk1-f199.google.com (mail-qk1-f199.google.com
 [209.85.222.199]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-392-Lu4sUoE0NGSUP-C6W9gisQ-1; Tue, 03 Dec 2019 11:59:13 -0500
Received: by mail-qk1-f199.google.com with SMTP id e23so2579620qka.23
        for <kvm@vger.kernel.org>; Tue, 03 Dec 2019 08:59:13 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=X1aOoFKR3ey7vYZ+BVQXR0mg8IAiU6yM63K4W6de1EU=;
        b=lykTzv7FXLqLCINlGEk2KF1BPUfyKRoolC6L85f5A88GyzBIoDXm7lSs2ZEVFy/e1u
         IDZNvHnj4qsoFwr2queKAw0Cve6qF8KOvQUbUQlToTG0/xjcAUpYPHXeIJrnsr581j71
         Sc71AKkJ6KMrvaaK648+4u4nkFFX6X2VVt/s3qatHOTHDT3vUrfxvHRfuHzqbwCaW7RL
         7lfctvovQ+3vordEm9/xtmGAnRfQ0OS8DlsNeUdwfakag8DhnJa/dXrO05dHCMbk7RhJ
         ktNE25sBCVC4bpCXwreba/6GK5kFwztp8o7FmfMRnPdmfFldmIv9cB4F2Ncd+P13DqV7
         ZAzQ==
X-Gm-Message-State: APjAAAWR6u3ED2um4Kb0DMsN8d/NMwO6GCbUp/7pDB44XKSyGOyD+L+t
        UhlkWZw54LuUvVsO938nVqm+u5blzKBta4sjVqpbxzuAHFutl+dJKpUap7vILLmD3gixeOtwXlc
        VWuckdBSLjx0J
X-Received: by 2002:a05:6214:6ad:: with SMTP id
 s13mr6005916qvz.208.1575392352968;
        Tue, 03 Dec 2019 08:59:12 -0800 (PST)
X-Google-Smtp-Source: 
 APXvYqzGh7RoR7bXUGd1EORpILas9d+Vrxhpz2zfR0xOT0S88RHqKjgS/dHqRB0sUoTvPCbET7czXw==
X-Received: by 2002:a05:6214:6ad:: with SMTP id
 s13mr6005894qvz.208.1575392352740;
        Tue, 03 Dec 2019 08:59:12 -0800 (PST)
Received: from xz-x1.yyz.redhat.com ([104.156.64.74])
        by smtp.gmail.com with ESMTPSA id
 a16sm482585qkn.48.2019.12.03.08.59.11
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Dec 2019 08:59:11 -0800 (PST)
From: Peter Xu <peterx@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Cc: Nitesh Narayan Lal <nitesh@redhat.com>,
        Paolo Bonzini <pbonzini@redhat.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        peterx@redhat.com, Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [PATCH v4 5/6] KVM: X86: Fix callers of kvm_apic_match_dest() to use
 correct macros
Date: Tue,  3 Dec 2019 11:59:02 -0500
Message-Id: <20191203165903.22917-6-peterx@redhat.com>
X-Mailer: git-send-email 2.21.0
In-Reply-To: <20191203165903.22917-1-peterx@redhat.com>
References: <20191203165903.22917-1-peterx@redhat.com>
MIME-Version: 1.0
X-MC-Unique: Lu4sUoE0NGSUP-C6W9gisQ-1
X-Mimecast-Spam-Score: 0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Callers of kvm_apic_match_dest() should always pass in APIC_DEST_*
macros for either dest_mode and short_hand parameters.  Fix up all the
callers of kvm_apic_match_dest() that are not following the rule.

Reported-by: Sean Christopherson <sean.j.christopherson@intel.com>
Reported-by: Vitaly Kuznetsov <vkuznets@redhat.com>
Reviewed-by: Vitaly Kuznetsov <vkuznets@redhat.com>
Signed-off-by: Peter Xu <peterx@redhat.com>
---
 arch/x86/kvm/ioapic.c   | 11 +++++++----
 arch/x86/kvm/irq_comm.c |  3 ++-
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/arch/x86/kvm/ioapic.c b/arch/x86/kvm/ioapic.c
index e623a4f8d27e..f53daeaaeb37 100644
--- a/arch/x86/kvm/ioapic.c
+++ b/arch/x86/kvm/ioapic.c
@@ -108,8 +108,9 @@ static void __rtc_irq_eoi_tracking_restore_one(struct kvm_vcpu *vcpu)
 	union kvm_ioapic_redirect_entry *e;
 
 	e = &ioapic->redirtbl[RTC_GSI];
-	if (!kvm_apic_match_dest(vcpu, NULL, 0,	e->fields.dest_id,
-				e->fields.dest_mode))
+	if (!kvm_apic_match_dest(vcpu, NULL, APIC_DEST_NOSHORT,
+				 e->fields.dest_id,
+				 kvm_lapic_irq_dest_mode(!!e->fields.dest_mode)))
 		return;
 
 	new_val = kvm_apic_pending_eoi(vcpu, e->fields.vector);
@@ -250,8 +251,10 @@ void kvm_ioapic_scan_entry(struct kvm_vcpu *vcpu, ulong *ioapic_handled_vectors)
 		if (e->fields.trig_mode == IOAPIC_LEVEL_TRIG ||
 		    kvm_irq_has_notifier(ioapic->kvm, KVM_IRQCHIP_IOAPIC, index) ||
 		    index == RTC_GSI) {
-			if (kvm_apic_match_dest(vcpu, NULL, 0,
-			             e->fields.dest_id, e->fields.dest_mode) ||
+			u16 dm = kvm_lapic_irq_dest_mode(!!e->fields.dest_mode);
+
+			if (kvm_apic_match_dest(vcpu, NULL, APIC_DEST_NOSHORT,
+						e->fields.dest_id, dm) ||
 			    kvm_apic_pending_eoi(vcpu, e->fields.vector))
 				__set_bit(e->fields.vector,
 					  ioapic_handled_vectors);
diff --git a/arch/x86/kvm/irq_comm.c b/arch/x86/kvm/irq_comm.c
index 22108ed66a76..7d083f71fc8e 100644
--- a/arch/x86/kvm/irq_comm.c
+++ b/arch/x86/kvm/irq_comm.c
@@ -417,7 +417,8 @@ void kvm_scan_ioapic_routes(struct kvm_vcpu *vcpu,
 
 			kvm_set_msi_irq(vcpu->kvm, entry, &irq);
 
-			if (irq.level && kvm_apic_match_dest(vcpu, NULL, 0,
+			if (irq.level &&
+			    kvm_apic_match_dest(vcpu, NULL, APIC_DEST_NOSHORT,
 						irq.dest_id, irq.dest_mode))
 				__set_bit(irq.vector, ioapic_handled_vectors);
 		}

From patchwork Tue Dec  3 16:59:03 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Xu <peterx@redhat.com>
X-Patchwork-Id: 11271703
Return-Path: <SRS0=eZxF=ZZ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 8D2A0112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:21 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 6BC632073F
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  3 Dec 2019 16:59:21 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="FsgU13+I"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727312AbfLCQ7U (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 3 Dec 2019 11:59:20 -0500
Received: from us-smtp-delivery-1.mimecast.com ([207.211.31.120]:38657 "EHLO
        us-smtp-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org
        with ESMTP id S1727287AbfLCQ7T (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 3 Dec 2019 11:59:19 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1575392358;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=CokILJJqXdrHsv0tjIebaIYM+c7ZHf6WdfuYQQUg9IQ=;
        b=FsgU13+IIBIcOL+HuIboMe54DBKzdFnlf38fswG+IvCmvipPkt2j8cdrFub+RvznYBF59k
        qxrsg4A1kb+2Hbb2kVo1ZBMbQvST22tN/Fdp8PnldYm/R5iF+8TbtpiuHTqExObFBGb3no
        i77ezpnKUzi2qCs+sCmMfLnTKquB3zY=
Received: from mail-qv1-f72.google.com (mail-qv1-f72.google.com
 [209.85.219.72]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-11-dpJotGpbMdy_N_UBCgOtrw-1; Tue, 03 Dec 2019 11:59:15 -0500
Received: by mail-qv1-f72.google.com with SMTP id g33so2588945qvd.7
        for <kvm@vger.kernel.org>; Tue, 03 Dec 2019 08:59:15 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=rby7OAxJWRGIY59KJwhRvRc4bijIHbG4l5NvhzcAxK8=;
        b=dGRf/6UPgbM+p/W3w4CyJrmPAS0kY32jCUjUC8B3c+2ldjpyrQjWJYT6U4Wt4sTEf+
         euWORAlZ1Yyw1tz6qrjp8K5XURVZxxUy4LLlC7JQuLzcOtA312fOvleueEhZ5763Kq9E
         UBRVahiVToLNrqy8xk7UjGonkZPkN/NqB56Ug8qoTxsnCqQMz6UyGghOsbQ7yIcTjI0I
         yRAJCvu1Zc/8f+UqunIrIX3ES06yRuQB2LUj7npp0mnDW4rAnR+ktS6xcTAq+v+qNhiw
         utisgD3A7Fm1/o3g/wrCuN9qlkn5aPRODAbLYWA5RIg3y8N2Kn1/CPW3z/bhyHSfkClc
         /qDg==
X-Gm-Message-State: APjAAAXUwFo8x/1vUxIg0UTuL7U9ToojFFN5p09MWkw/0udv2FkB2fJJ
        bgnxiyKfMvHWjPKeWgTcOJt3NEwlFRgU2ARMLfwWXKMdzsbMLb/j82X7Ch6v0VHkYhtzuQP1xoV
        L6Pl9bwEOkG3t
X-Received: by 2002:a05:620a:2010:: with SMTP id
 c16mr5938044qka.386.1575392354612;
        Tue, 03 Dec 2019 08:59:14 -0800 (PST)
X-Google-Smtp-Source: 
 APXvYqzQfVoMDRqDnxNfwvnqyaXyIDMNqHRLocnfax32ITgCaGfDAssMZDEsbnRydHQ5WTxKScDc7A==
X-Received: by 2002:a05:620a:2010:: with SMTP id
 c16mr5938019qka.386.1575392354373;
        Tue, 03 Dec 2019 08:59:14 -0800 (PST)
Received: from xz-x1.yyz.redhat.com ([104.156.64.74])
        by smtp.gmail.com with ESMTPSA id
 a16sm482585qkn.48.2019.12.03.08.59.12
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Dec 2019 08:59:13 -0800 (PST)
From: Peter Xu <peterx@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Cc: Nitesh Narayan Lal <nitesh@redhat.com>,
        Paolo Bonzini <pbonzini@redhat.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        peterx@redhat.com, Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [PATCH v4 6/6] KVM: X86: Conert the last users of "shorthand = 0" to
 use macros
Date: Tue,  3 Dec 2019 11:59:03 -0500
Message-Id: <20191203165903.22917-7-peterx@redhat.com>
X-Mailer: git-send-email 2.21.0
In-Reply-To: <20191203165903.22917-1-peterx@redhat.com>
References: <20191203165903.22917-1-peterx@redhat.com>
MIME-Version: 1.0
X-MC-Unique: dpJotGpbMdy_N_UBCgOtrw-1
X-Mimecast-Spam-Score: 0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Change the last users of "shorthand = 0" to use APIC_DEST_NOSHORT.

Signed-off-by: Peter Xu <peterx@redhat.com>
---
 arch/x86/kvm/ioapic.c   | 4 ++--
 arch/x86/kvm/irq_comm.c | 2 +-
 arch/x86/kvm/x86.c      | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/x86/kvm/ioapic.c b/arch/x86/kvm/ioapic.c
index f53daeaaeb37..77538fd77dc2 100644
--- a/arch/x86/kvm/ioapic.c
+++ b/arch/x86/kvm/ioapic.c
@@ -330,7 +330,7 @@ static void ioapic_write_indirect(struct kvm_ioapic *ioapic, u32 val)
 		if (e->fields.delivery_mode == APIC_DM_FIXED) {
 			struct kvm_lapic_irq irq;
 
-			irq.shorthand = 0;
+			irq.shorthand = APIC_DEST_NOSHORT;
 			irq.vector = e->fields.vector;
 			irq.delivery_mode = e->fields.delivery_mode << 8;
 			irq.dest_id = e->fields.dest_id;
@@ -379,7 +379,7 @@ static int ioapic_service(struct kvm_ioapic *ioapic, int irq, bool line_status)
 	irqe.trig_mode = entry->fields.trig_mode;
 	irqe.delivery_mode = entry->fields.delivery_mode << 8;
 	irqe.level = 1;
-	irqe.shorthand = 0;
+	irqe.shorthand = APIC_DEST_NOSHORT;
 	irqe.msi_redir_hint = false;
 
 	if (irqe.trig_mode == IOAPIC_EDGE_TRIG)
diff --git a/arch/x86/kvm/irq_comm.c b/arch/x86/kvm/irq_comm.c
index 7d083f71fc8e..9d711c2451c7 100644
--- a/arch/x86/kvm/irq_comm.c
+++ b/arch/x86/kvm/irq_comm.c
@@ -121,7 +121,7 @@ void kvm_set_msi_irq(struct kvm *kvm, struct kvm_kernel_irq_routing_entry *e,
 	irq->msi_redir_hint = ((e->msi.address_lo
 		& MSI_ADDR_REDIRECTION_LOWPRI) > 0);
 	irq->level = 1;
-	irq->shorthand = 0;
+	irq->shorthand = APIC_DEST_NOSHORT;
 }
 EXPORT_SYMBOL_GPL(kvm_set_msi_irq);
 
diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 3b00d662dc14..f6d778436e15 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -7355,7 +7355,7 @@ static void kvm_pv_kick_cpu_op(struct kvm *kvm, unsigned long flags, int apicid)
 {
 	struct kvm_lapic_irq lapic_irq;
 
-	lapic_irq.shorthand = 0;
+	lapic_irq.shorthand = APIC_DEST_NOSHORT;
 	lapic_irq.dest_mode = APIC_DEST_PHYSICAL;
 	lapic_irq.level = 0;
 	lapic_irq.dest_id = apicid;
